test/_init.test.ts

```diff
@@ -0,0 +1,12 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ISuiteCallbackContext } from 'mocha';
+import { commands } from 'vscode';
+
+suite('Initialize extension', async function (this: ISuiteCallbackContext): Promise<void> {
+    this.timeout(120 * 1000); // two minutes should be enough time
+    await commands.executeCommand('azureFunctions.refresh'); //activate the extension before testing begins to make sure activation doesn't change any extensionVariables
```

Comment:
FYI, my solution was to simply use "if (!ext.tree) { ext.tree = new … }" kind of code in Extension.ts.

Reply:
Not sure that mine's better, though.

Reply:
I think I prefer Nathan's approach for the following reasons:
1. It doesn't require changes to production code
1. It automatically initializes/disposes a few things for us (i.e. extensionContext and outputChannel) rather than having to mock that stuff